pkg/netutil/subnet/subnet.go

```diff
@@ -19,11 +19,17 @@ package subnet
 import (
 	"fmt"
 	"net"
+
+	"github.com/containerd/nerdctl/v2/pkg/rootlessutil"
 )
 
 func GetLiveNetworkSubnets() ([]*net.IPNet, error) {
-	addrs, err := net.InterfaceAddrs()
-	if err != nil {
+	var addrs []net.Addr
+	if err := rootlessutil.WithDetachedNetNSIfAny(func() error {
+		var err2 error
+		addrs, err2 = net.InterfaceAddrs()
```

Comment:
Ah, I think this is what I was missing in my PR; I wasn't pointing to the right IP addresses, right? It's not an issue with `runc` as we discussed in Chicago.